ingestion/src/metadata/ingestion/source/pipeline/openlineage/metadata.py

```diff
@@ -0,0 +1,534 @@
+        :return:
+        """
+        try:
+            fields = table_input["facets"]["schema"]["fields"]
+
+            # @todo check if this way of passing type is ok
+            columns = [
+                Column(name=f.get("name"), dataType=f.get("type").upper())
+                for f in fields
+            ]
+            return columns
+        except KeyError:
+            return None
+
+    def get_create_table_request(self, table: Dict) -> Optional[Either]:
+        """
+        If certain table from Open Lineage events doesn't already exist in Open Metadata, register appropriate entity.
+        This makes sense especially for output facet of OpenLineage event - as database service ingestion is a scheduled
+        process we might fall into situation where we received Open Lineage event about creation of a table that is yet
+        to be ingested by database service ingestion process. To avoid missing on such lineage scenarios, we will create
+        table entity beforehand.
+
+        :param table: single object from inputs/outputs facet
+        :return: request to create the entity (if needed)
+        """
+        om_table_fqn = None
+
+        try:
+            table_details = OpenlineageSource._get_table_details(table)
+        except ValueError as e:
+            return Either(
+                left=StackTraceError(
+                    name="",
+                    error=f"Failed to get partial table name: {e}",
+                    stackTrace=traceback.format_exc(),
+                )
+            )
+        try:
+            om_table_fqn = self._get_table_fqn_from_om(table_details)
+
+            # if fqn found then it means table is already registered and we don't need to render create table request
+            return None
+        except FQNNotFoundException:
+            pass
+
+        # If OM Table FQN was not found based on OL Partial Name - we need to register it.
+        if not om_table_fqn:
+            try:
+                om_schema_fqn = self._get_schema_fqn_from_om(table_details.schema)
+            except FQNNotFoundException as e:
+                return Either(
+                    left=StackTraceError(
+                        name="",
+                        error=f"Failed to get fully qualified schema name: {e}",
+                        stackTrace=traceback.format_exc(),
+                    )
+                )
+
+            # After finding schema fqn (based on partial schema name) we know where we can create table
+            # and we move forward with creating request.
+            if om_schema_fqn:
+                columns = OpenlineageSource._get_om_table_columns(table) or []
+
+                request = CreateTableRequest(
+                    name=table_details.name,
+                    columns=columns,
+                    databaseSchema=om_schema_fqn,
+                )
+
+                return Either(right=request)
+        else:
+            pass
+
+        return None
```

Comment:
We only create table while ingesting the database service it self, creating table in pipeline metadata ingestion would not be a right approach if we are not able to find the table then we can just simply log this and skip the lineage processing.